### ISSUE TITLE
Add refreshPolicy for argocd application object a.k.a. (hard-)refresh-policy

### ISSUE THREAD

**Issue by drewboswell** [argoproj/argo-cd#9499]
# Summary

Add a refresh policy setting to argocd application objects (like syncPolicy, but specifically for refresh strategies).

# Motivation

This applies to many scenarios: tracking branches, head, overridden helm packages etc. I would like to set whether an argocd application does soft or hard refresh by default. Additionally setting the refresh related TTL from the app would also be useful.

### Example:
Rapid development in "snapshot" mode, override package until release and promotion is done.
--> this means that the oci helm coordinates stay the same, but the oci image changes. 
--> kubernetes deals with this by having an imagePullPolicy that you can set to Always, for constant hash verification and pull on difference.

**Comment by prkaturi** [comment]
Any update on this issue?

**Comment by cveld** [comment]
We have this workflow where the dev teams continuously pushes helm chart updates to the same registry reference e.g. `oci://myregistry.azurecr.io/helm/myrepo` keeping the same tag, e.g. `canary`. 
Argocd doesn't seem to check the hash of this reference, so it is not detecting any changes it needs to sync to the cluster.

With OP's request I understand this would be made possible to enforce a refresh policy on this object.

**Comment by cveld** [comment]
I don't believe `imagePullPolicy` set to `Always` enforces an always up-to-date container. You must also kick the pod or deployment.

**Comment by rdrgporto** [comment]
Hi, 

Any update of this issue? 

From my point of view, this type of behavior is often used quite a bit in a development environment.

Thanks in advance,

Regards

**Comment by andrii-korotkov-verkada** [comment]
Do you refer to the cases where the manifest stays the same, but some external resources change?
Why not just refresh the application manually through UI/CLI?

**Comment by mrivasxvale** [comment]
I have the same problem. 

If I push a helm chart to an OCI registry, if I push a new Chart to the same tag ArgoCd does not update the application.

**Comment by mrivasxvale** [comment]
> I have the same problem.
> 
> If I push a helm chart to an OCI registry, if I push a new Chart to the same tag ArgoCd does not update the application.

I managed to have it working, creating an repository of type `OCI` instead of `Helm` and using `oci://` in the the helm source
